spring-core-4/src/main/java/com/baeldung/dynamic/autowire/InterfaceDynamicAutowireService.java

```diff
@@ -0,0 +1,26 @@
+package com.baeldung.dynamic.autowire;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class InterfaceDynamicAutowireService {
```

Comment:
why interface in the name?

Reply:
Wanted to show in the name, that this solution is based on interface autowiring, will change for a better name because this one can be missleading